owncloud/owncloud.py

```diff
@@ -878,6 +919,69 @@ def remove_user_from_group(self, user_name, group_name):
 
         raise ResponseError(res)
 
+    def add_user_to_subadminGroup(self, user_name, group_name):
+        """Adds a user to a subadmin group.
+
+        :param user_name:  name of user to be added to subadmin group
+        :param group_name:  name of subadmin group
+        :returns: True if user added
+        :raises: ResponseError in case an HTTP error status was returned
+
+        """
+
+        res = self.__make_ocs_request(
+            'POST',
+            self.OCS_SERVICE_CLOUD,
+            'users/' + user_name + '/subadmins',
```

Comment:
The regex for creating groups is here: https://github.com/owncloud/core/blob/master/apps/provisioning_api/lib/groups.php#L65. When this was written, the OC_User class used to just fail on errors and since we needed something nicer the validation was copied here. I'm guessing the rules have changed, but the validation still remains in here.